src/SearchResults/index.js

```diff
@@ -682,6 +692,70 @@ function vanillaSortFn(order, data) {
   return data.sort(order);
 }
 
+/**
+ * @typedef FacetOrdering
+ * @type {Object}
+ * @property {string[]} [order]
+ * @property {'count' | 'alpha' | 'hidden'} [sortRemainingBy]
+ */
+
+/**
+ * Sorts facet arrays via their facet ordering
+ * @param {Array} facetValues the values
+ * @param {FacetOrdering} facetOrdering the ordering
+ * @returns {Array}
+ */
+function sortViaFacetOrdering(facetValues, facetOrdering) {
+  var orderedFacets = [];
+  var remainingFacets = [];
+
+  var order = facetOrdering.order || [];
+  /**
+   * an object with the keys being the values in order, the values their index:
+   * ['one', 'two'] -> { one: 0, two: 1 }
+   */
+  var reverseOrder = order.reduce(function(acc, name, i) {
+    acc[name] = i;
+    return acc;
+  }, {});
+
+  facetValues.forEach(function(item) {
+    if (reverseOrder[item.name] !== undefined) {
+      orderedFacets[reverseOrder[item.name]] = item;
+    } else {
+      remainingFacets.push(item);
+    }
+  });
+
+  var sortRemainingBy = facetOrdering.sortRemainingBy;
+  var ordering;
+  if (sortRemainingBy === 'hidden') {
+    return orderedFacets;
+  } else if (sortRemainingBy === 'alpha') {
+    ordering = [['name'], ['asc']];
+  } else {
+    ordering = [['count'], ['desc']];
+  }
```

Comment:
Making sure: we decide to fallback to sorting by count and not throwing if the provided order is wrong?

Reply:
it's something that's a bit blurry at the moment, the spec mentions falling back to sortFacetValuesBy, but I can't retrieve that in the result-side, so I think falling back to just count by default is reasonable.

Alternatively we could make sortRemainingBy required in the dashboard possibly. What did you pick @VladislavFitz?

Reply:
this is about the key sortRemainingBy, what do you sort by if the key isn't given?

Reply:
Sort by count by default sounds good to me 👍